File: src/features/objectDetection/domain/detectionResult.ts
/**
 * Модель результата детекции объекта.
 */

export interface DetectionResult {
  /** Название объекта (из label map) */
  label: string;

  /** Уверенность модели (0–1) */
  confidence: number;

  /** Bounding box: [x, y, width, height] в пикселях исходного изображения */
  boundingBox: {
    x: number;
    y: number;
    width: number;
    height: number;
  };
}

File: lib/features/object_detection/domain/detection_result.dart
import 'package:equatable/equatable.dart';

/// Результат детекции одного объекта (§8.2 ТЗ).
class DetectionResult extends Equatable {
  const DetectionResult({
    required this.label,
    required this.confidence,
    required this.boundingBox,
  });

  /// Человекочитаемое название объекта (например, «бутылка»).
  final String label;

  /// Уверенность модели (0.0 – 1.0).
  final double confidence;

  /// Bounding box: [left, top, right, bottom] в нормализованных координатах.
  final List<double> boundingBox;

  @override
  List<Object?> get props => [label, confidence, boundingBox];
}
